swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = inventory or []
+
+    def add(self, added_item):
+        (self.inventory).append(added_item)
+        return added_item
+    
+    def remove(self,item_removed):
+        if item_removed not in self.inventory:
+            return False
+        else:
+            for item in self.inventory:
+                if item_removed == item:
+                    (self.inventory).remove(item)
+                    return item_removed
+
+    def get_by_id(self, id_num):
+        for item in self.inventory:
+            if item.id == id_num:
+                return item
+            
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        else:
+            self.add(their_item)
+            other_vendor.add(my_item)
+            self.remove(my_item)
+            other_vendor.remove(their_item)
+            return True
+        
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+        else:
+            self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+            return True
+        
+    def get_by_category(self, category_type):
+        items_in_category = []
+        for item in self.inventory:
+            if category_type == item.get_category():
+                items_in_category.append(item)
+        return items_in_category
```

Comment:
Notice this pattern of: 
```python
result_list = []
for element in source_list:
    if some_condition(element):
        result_list.append(element)
```

can be rewritten using a list comprehension as:
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like:
```python
items_in_category = [item for item in self.inventory if item.get_category() == category_type]
return items_in_category
```

At first, this may seem more difficult to read, but comprehensions are a very common python style, so I encourage y’all to try working with them!